execution/executor/src/workflow/do_get_execution_output.rs

```diff
@@ -49,7 +49,8 @@ impl DoGetExecutionOutput {
         transactions: ExecutableTransactions,
         state_view: CachedStateView,
         onchain_config: BlockExecutorConfigFromOnchain,
-        append_state_checkpoint_to_block: Option<HashValue>,
```

Comment:
It will clear caches for now, but I 100% agree. Ideally, we want versions for chunks (and things like aptos-debugger, and we can naturally compare if previous end = current begin), that we hide under enum. Will address this on top of the PR stack 👍

Reply:
I am wondering if you need a separate boolean for append_state_checkpoint_to_block. maybe if it is equivalent - a comment that says state checkpoint will be appended in case that current_block.is_some()

Reply:
Added a comment👍